agoric/src/httpSender.ts

```diff
@@ -0,0 +1,16 @@
+import { Requester } from '@chainlink/external-adapter'
+
+export interface Action {
+  type: string
+  data: unknown
+}
+export interface HTTPSenderReply {
+  status: number
+  response: unknown
+}
+export type HTTPSender = (obj: Action) => Promise<HTTPSenderReply>
+
+export const makeHTTPSender = (url: string): HTTPSender => async (data) => {
```

Comment:
I don't think `makeHTTPSender` is really necessary, it's just wrapping the oracle url. I recommend you to follow the [external adapter example](https://github.com/smartcontractkit/external-adapters-js/tree/develop/example/src). You will see that all this configuration can be set under `config`.